From 9480360bc8f522f0bb83cf3553b407b9d649e9f2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Fri, 19 Jul 2019 16:26:02 +0200
Subject: [PATCH 02/71] py3: another warning about invalid escape

---
 src/tinycss/token_data.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tinycss/token_data.py b/src/tinycss/token_data.py
index 40bff86e77..dfbafc1781 100644
--- a/src/tinycss/token_data.py
+++ b/src/tinycss/token_data.py
@@ -214,7 +214,7 @@ FIND_NEWLINES = lambda x : list(re.compile(COMPILED_MACROS['nl']).finditer(x))
 
 
 class Token(object):
-    """A single atomic token.
+    r"""A single atomic token.
 
     .. attribute:: is_container
 
