From f39eff0117e1b789caad121e79b193b38fed8cf7 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Mon, 22 Jul 2019 11:20:50 -0400
Subject: [PATCH 09/71] unicode_check: do not try to check pyuic-generated
 files

---
 setup/port.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup/port.py b/setup/port.py
index ecd09e2e9e..401764fcb7 100644
--- a/setup/port.py
+++ b/setup/port.py
@@ -72,7 +72,8 @@ class Base(Command):
     def get_files(self):
         from calibre import walk
         for path in walk(os.path.join(self.SRC, 'calibre')):
-            if path.endswith('.py') and not os.path.basename(path) in self.EXCLUDED_BASENAMES:
+            if (path.endswith('.py') and not path.endswith('_ui.py') and not
+                    os.path.basename(path) in self.EXCLUDED_BASENAMES):
                 yield path
 
     def file_hash(self, f):
