From 2a7f3265b0df583db5ee605f90846c4f9110a279 Mon Sep 17 00:00:00 2001
From: Kovid Goyal <kovid@kovidgoyal.net>
Date: Sun, 18 Aug 2019 13:03:45 +0530
Subject: [PATCH 68/71] Update derStandaard

---
 recipes/der_standard.recipe | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/recipes/der_standard.recipe b/recipes/der_standard.recipe
index bb0d198ea6..0663120b17 100644
--- a/recipes/der_standard.recipe
+++ b/recipes/der_standard.recipe
@@ -61,6 +61,7 @@ class DerStandardRecipe(BasicNewsRecipe):
 
     def get_browser(self):
         br = BasicNewsRecipe.get_browser(self)
+        br.set_simple_cookie('DSGVO_Check', '', '.derstandard.at')
         headers = {
             'X-Requested-With': 'XMLHttpRequest',
             'Content-Type': 'application/json; charset=UTF-8',
@@ -69,12 +70,12 @@ def get_browser(self):
             'Cache-Control': 'no-cache'
         }
         import mechanize
-        req =   mechanize.Request(url='https://derstandard.at/privacyprotection/api/agree', data=None, headers=headers, method='POST')
+        req = mechanize.Request(url='https://apps.derstandard.at/privacyprotection/api/agree', data=None, headers=headers, method='POST')
         br.open(req)
         return br
 
     keep_only_tags = [
-        classes('artikel'),
+        classes('article-header article-body article-origins article-subtitle article-pubdate'),
     ]
 
     remove_tags = [
