--- calibre/src/calibre/ebooks/pdf/reflow.cpp	2012-05-17 14:44:05.000000000 +0200
+++ calibre/src/calibre/ebooks/pdf/reflow.cpp	2012-05-17 14:54:04.000000000 +0200
@@ -648,7 +648,7 @@ void XMLOutputDev::process_link(AnnotLin
 
 
 void XMLOutputDev::endPage() {
-    Links *slinks = catalog->getPage(current_page->number())->getLinks(catalog);
+    Links *slinks = catalog->getPage(current_page->number())->getLinks();
     for (int i = 0; i < slinks->getNumLinks(); i++)
     {
         this->process_link(slinks->getLink(i));
@@ -879,7 +879,7 @@ vector<char>* Reflow::render_first_page(
         throw ReflowException("Failed to allocate SplashOutputDev");
     }
     try {
-        out->startDoc(doc->getXRef());
+        out->startDoc(doc);
         out->startPage(1, NULL);
 
         double pg_w, pg_h;
