diff -Nur calibre.orig/src/calibre/gui2/ui.py calibre/src/calibre/gui2/ui.py
--- calibre.orig/src/calibre/gui2/ui.py	2014-08-19 15:21:03.631159912 -0600
+++ calibre/src/calibre/gui2/ui.py	2014-08-19 15:21:47.033416065 -0600
@@ -883,7 +883,6 @@
         if write_settings:
             self.write_settings()
         self.check_messages_timer.stop()
-        self.update_checker.terminate()
         self.listener.close()
         self.job_manager.server.close()
         self.job_manager.threaded_server.close()
diff -Nur calibre.orig/src/calibre/gui2/update.py calibre/src/calibre/gui2/update.py
--- calibre.orig/src/calibre/gui2/update.py	2014-08-19 15:21:03.632159894 -0600
+++ calibre/src/calibre/gui2/update.py	2014-08-19 15:22:25.305760137 -0600
@@ -136,11 +136,6 @@
 
     def __init__(self, opts):
         self.last_newest_calibre_version = NO_CALIBRE_UPDATE
-        if not opts.no_update_check:
-            self.update_checker = CheckForUpdates(self)
-            self.update_checker.update_found.connect(self.update_found,
-                    type=Qt.QueuedConnection)
-            self.update_checker.start()
 
     def recalc_update_label(self, number_of_plugin_updates):
         self.update_found(self.last_newest_calibre_version, number_of_plugin_updates)
