diff -up calibre/setup/install.py.executables calibre/setup/install.py
--- calibre/setup/install.py.executables	2010-01-06 11:12:45.214637969 +0200
+++ calibre/setup/install.py	2010-01-06 11:12:58.056763017 +0200
@@ -185,7 +185,7 @@ class Develop(Command):
         script = template.format(
                 module=mod, func=func,
                 path=self.libdir, resources=self.sharedir,
-                executables=self.staging_bindir,
+                executables=self.bindir,
                 extensions=self.j(self.libdir, 'calibre', 'plugins'))
         path = self.j(self.staging_bindir, name)
         if not os.path.exists(self.staging_bindir):
