diff -up calibre/src/calibre/gui2/main.py.no_update calibre/src/calibre/gui2/main.py
--- calibre/src/calibre/gui2/main.py.no_update	2009-11-16 14:21:55.200387171 +0200
+++ calibre/src/calibre/gui2/main.py	2009-11-16 14:22:10.400510757 +0200
@@ -221,11 +221,11 @@ class Main(MainWindow, Ui_MainWindow, De
         self.latest_version = ' '
         self.vanity.setText(self.vanity_template%dict(version=' ', device=' '))
         self.device_info = ' '
-        if not opts.no_update_check:
-            self.update_checker = CheckForUpdates()
-            QObject.connect(self.update_checker,
-                    SIGNAL('update_found(PyQt_PyObject)'), self.update_found)
-            self.update_checker.start()
+        # if not opts.no_update_check:
+        #     self.update_checker = CheckForUpdates()
+        #     QObject.connect(self.update_checker,
+        #             SIGNAL('update_found(PyQt_PyObject)'), self.update_found)
+        #     self.update_checker.start()
         ####################### Status Bar #####################
         self.status_bar = StatusBar(self.jobs_dialog, self.system_tray_icon)
         self.setStatusBar(self.status_bar)
