diff -Nur calibre.orig/recipes/economist_free.recipe calibre/recipes/economist_free.recipe
--- calibre.orig/recipes/economist_free.recipe	2012-08-09 21:33:57.000000000 -0600
+++ calibre/recipes/economist_free.recipe	2012-08-13 21:01:07.567981819 -0600
@@ -185,7 +185,7 @@
         lambda x:'</html>')]
 
     def parse_index(self):
-        from calibre.web.feeds.feedparser import parse
+        from feedparser import parse
         if self.test:
             self.oldest_article = 14.0
         raw = self.index_to_soup(
diff -Nur calibre.orig/recipes/lenta_ru.recipe calibre/recipes/lenta_ru.recipe
--- calibre.orig/recipes/lenta_ru.recipe	2012-08-09 21:33:57.000000000 -0600
+++ calibre/recipes/lenta_ru.recipe	2012-08-13 21:01:19.778043630 -0600
@@ -4,7 +4,7 @@
 Lenta.ru
 '''
 
-from calibre.web.feeds.feedparser import parse
+from feedparser import parse
 from calibre.ebooks.BeautifulSoup import Tag
 from calibre.web.feeds.news import BasicNewsRecipe
 import re
diff -Nur calibre.orig/recipes/vedomosti.recipe calibre/recipes/vedomosti.recipe
--- calibre.orig/recipes/vedomosti.recipe	2012-08-09 21:33:57.000000000 -0600
+++ calibre/recipes/vedomosti.recipe	2012-08-13 21:01:27.924085824 -0600
@@ -4,7 +4,7 @@
 Ведомости
 '''
 
-from calibre.web.feeds.feedparser import parse
+from feedparser import parse
 from calibre.ebooks.BeautifulSoup import Tag
 from calibre.web.feeds.news import BasicNewsRecipe
 
