diff -Nur calibre.orig/src/calibre/gui2/update.py calibre/src/calibre/gui2/update.py
--- calibre.orig/src/calibre/gui2/update.py	2014-08-22 13:38:27.921743385 -0600
+++ calibre/src/calibre/gui2/update.py	2014-08-22 13:50:47.003149256 -0600
@@ -56,20 +56,6 @@
         while True:
             calibre_update_version = NO_CALIBRE_UPDATE
             plugins_update_found = 0
-            try:
-                version = get_newest_version()
-                if version[:2] > numeric_version[:2]:
-                    calibre_update_version = version
-            except Exception as e:
-                prints('Failed to check for calibre update:', as_unicode(e))
-            try:
-                update_plugins = get_plugin_updates_available(raise_error=True)
-                if update_plugins is not None:
-                    plugins_update_found = len(update_plugins)
-            except Exception as e:
-                prints('Failed to check for plugin update:', as_unicode(e))
-            if calibre_update_version != NO_CALIBRE_UPDATE or plugins_update_found > 0:
-                self.update_found.emit(calibre_update_version, plugins_update_found)
             self.sleep(self.INTERVAL)
 
 class UpdateNotification(QDialog):
