diff -Nur calibre.orig/src/calibre/ebooks/pdf/reflow.cpp calibre/src/calibre/ebooks/pdf/reflow.cpp
--- calibre.orig/src/calibre/ebooks/pdf/reflow.cpp	2011-07-15 10:54:35.000000000 -0600
+++ calibre/src/calibre/ebooks/pdf/reflow.cpp	2011-07-16 10:01:28.576025342 -0600
@@ -625,7 +625,7 @@
   return oss.str();
 }
 
-void XMLOutputDev::process_link(Link* link){
+void XMLOutputDev::process_link(AnnotLink* link){
   double _x1, _y1, _x2, _y2;
   int x1, y1, x2, y2;
   
diff -Nur calibre.orig/src/calibre/ebooks/pdf/reflow.h calibre/src/calibre/ebooks/pdf/reflow.h
--- calibre.orig/src/calibre/ebooks/pdf/reflow.h	2011-07-15 10:54:35.000000000 -0600
+++ calibre/src/calibre/ebooks/pdf/reflow.h	2011-07-16 10:00:49.766268969 -0600
@@ -244,6 +244,6 @@
     XMLImages *images;
     PDFDoc *doc;
 
-    void process_link(Link* link);
+    void process_link(AnnotLink* link);
 };
 }
